sdk/keyvault/azcertificates/models.go

```diff
@@ -101,8 +101,35 @@ func certificateAttributesFromGenerated(g *generated.CertificateAttributes) *Cer
 	}
 }
 
-// Certificate - A certificate bundle consists of a certificate (X509) plus its properties.
-type Certificate struct {
+// KeyVaultCertificate - A certificate bundle consists of a certificate (X509) plus its properties.
+type KeyVaultCertificate struct {
+	// The certificate attributes.
+	Properties *CertificateProperties `json:"attributes,omitempty"`
+
+	// CER contents of x509 certificate.
+	Cer []byte `json:"cer,omitempty"`
+
+	// The content type of the secret. eg. 'application/x-pem-file' or 'application/x-pkcs12',
+	ContentType *string `json:"contentType,omitempty"`
+
+	// Application specific metadata in the form of key-value pairs
+	Tags map[string]string `json:"tags,omitempty"`
+
+	// READ-ONLY; The certificate id.
+	ID *string `json:"id,omitempty" azure:"ro"`
+
+	// READ-ONLY; The key ID.
+	KeyID *string `json:"kid,omitempty" azure:"ro"`
+
+	// READ-ONLY; The secret ID.
+	SecretID *string `json:"sid,omitempty" azure:"ro"`
+
+	// READ-ONLY; Thumbprint of the certificate.
+	X509Thumbprint []byte `json:"x5t,omitempty" azure:"ro"`
+}
+
+// KeyVaultCertificateWithPolicy - A certificate bundle consists of a certificate (X509) with a policy, and its properties.
+type KeyVaultCertificateWithPolicy struct {
```

Comment:
Nit: wouldn't it make sense to just declare `KeyVaultCertificate` as a nested struct e.g.

```go
type KeyVaultCertificate struct {
  // ...
}

type KeyVAultCertificteWithPolicy struct {
  KeyVaultCertificate

  // READ-ONLY; The management policy.
  Policy *CertificatePolicy `json:"policy,omitempty" azure:"ro"`
}
```

It's derived from KeyVaultCertificate in .NET and Java, and practically in other languages.

Reply:
Composition makes it painful when creating struct literals of these things due to the extra nesting.  It can also introduce some unwanted side-effects by promoting methods on `KeyVaultCertificate` to the parent.

Reply:
Understood, but in this case please at least consider:

1. These models are intended as a derivation, so members appearing in the `KeyVaultCertificateWithPolicy` model from `KeyVaultCertificate` are indeed intended.
2. This model shouldn't be creatable by the client anyway, so construction woes with struct literals (thanks, I forgot the name) shouldn't be much of a concern.

That said, I signed off. Merely an idea/nit, and certain up to your idiomatic designs to decide. 🙂